src/consumer_queue.cpp

```diff
@@ -1,13 +1,23 @@
 #include "consumer_queue.h"
 #include "common.h"
-#include "sample.h"
 #include "send_buffer.h"
 #include <chrono>
 
 using namespace lsl;
 
-consumer_queue::consumer_queue(std::size_t max_capacity, send_buffer_p registry)
-	: registry_(registry), buffer_(max_capacity) {
+consumer_queue::consumer_queue(std::size_t size, send_buffer_p registry)
+		: registry_(registry),
+		  buffer_(new item_t[size]),
+		  size_(size),
+		  // largest integer at which we can wrap correctly
+		  wrap_at_(std::numeric_limits<std::size_t>::max() - size - std::numeric_limits<std::size_t>::max() % size)
+{
+	assert(size_ > 1);
+	for (std::size_t i=0; i<size_; ++i)
```

Comment:
Do we need to `store()` everything in the constructor already could we use standard initialization (`: write_idx_(0), read_idx_(0)`…)?

Reply:
Before I looked it up, I was actually under the impression that write_idx_(0) would use the default memory order of `std::atomic` when it's used like a plain value, which is `memory_order_seq_cst`, and that's quite a bit stronger than what this data structure needs (so I'd worry a bit about some compiler pessimizations, e.g., additional memory fences). However, it turns out that that type of initialization is apparently [non-atomic](https://en.cppreference.com/w/cpp/atomic/atomic/atomic), so that'd mean that the explicit store is actually necessary (since another thread can later acquire the value).

Reply:
What about an `std::atomic_thread_fence(std::memory_order_release);` at the end of the constructor?